src/scaledmap.jl

```diff
@@ -0,0 +1,105 @@
+#=
+scaledmap.jl
+Efficiently handle "scale * A" when the scale factor is Real or Complex.
+In principle, the map A itself could be any type,
+but for simplicity this code focuses "solely" on Real or Complex LinearMaps.
+=#
+
+using LinearAlgebra: UniformScaling
+import LinearAlgebra: issymmetric, ishermitian, isposdef
+
+#=
+todo: i would prefer to let "scale" be a different type than the map.
+e.g., if A is ComplexF32, allow scale to be Real so that multiplying
+by scale would require only a real multiply.  but i could not figure
+out how to get @inferred to pass when i had scale::S
+=#
+
+"""
+    ScaledMap{T, A<:LinearMapRC} <: LinearMap{T}
+"""
+struct ScaledMap{T, A<:LinearMapRC} <: LinearMap{T}
+    scale::T
+    map::A
+    function ScaledMap{T, A}(scale::S, map::A) where {T, S <: RealOrComplex, A <: LinearMapRC}
+        promote_type(S, eltype(map)) == T || throw(InexactError())
+        new{T, A}(scale, map)
+    end
+end
+
+# constructor
+ScaledMap(scale::S, map::A) where {S <: RealOrComplex, A <: LinearMapRC} =
+    ScaledMap{promote_type(S,eltype(map)), A}(scale, map)
+
+# show
+function Base.show(io::IO, A::ScaledMap{T}) where {T}
+    println(io, "LinearMaps.ScaledMap{$T}, scale = $(A.scale)")
+    show(io, A.map)
+end
+
+# basic methods
+Base.size(A::ScaledMap) = size(A.map)
+Base.isreal(A::ScaledMap) = isreal(A.scale) && isreal(A.map)
+LinearAlgebra.issymmetric(A::ScaledMap) = isreal(A.scale) && issymmetric(A.map)
+LinearAlgebra.ishermitian(A::ScaledMap) = ishermitian(A.map)
+
+# subtle issue here that if map is Complex, then so is scale
+LinearAlgebra.isposdef(A::ScaledMap) =
+    isposdef(A.map) && isreal(A.scale) && real(A.scale) > 0
+
+Base.transpose(A::ScaledMap) = ScaledMap(A.scale, transpose(A.map))
+Base.adjoint(A::ScaledMap) = ScaledMap(conj(A.scale), adjoint(A.map))
+
+# comparison (sufficient, not necessary)
+Base.:(==)(A::ScaledMap, B::ScaledMap) =
+    (eltype(A) == eltype(B) && A.map == B.map) && A.scale == B.scale
+
+# approximate comparison (because == for real scalars is dubious)
+Base.:(≈)(A::ScaledMap, B::ScaledMap) =
+    (eltype(A) == eltype(B) && A.map == B.map) && A.scale ≈ B.scale
+
+# x * conj(x) is real in math, but not perfectly so in computation,
+# but we want it to be real here (when possible) for isposdef()
+# and so that α*conj(α)*A is has a real scale when A is Real
+@inline _scalar_product(a, b) = isreal(a*b) ? real(a*b) : a*b
```

Comment:
Oh yeah, that was something I added when I was stuck on how to handle `isposdef` but later I resolve that and never went back to remove that.  It was a little bit tricky because you can have `λ::S = 3 + 0im` which is still fine for `isposdef`.

Reply:
I understand that one may have `imag(x*conj(x))` not exactly equal to zero. But I don't see anywhere in `Base` or `LinearAlgebra` that people would try to guess that configuration and make the result real manually (at the expense of type instability). There is still your test for `isposdef`, and it passes because it happens that the imaginary part is exactly equal to 0.

Reply:
Oh, I didn't think of using `isposdef(scale)` -- that's elegant!
I was curious how Julia handles it and the code is in `dense.jl` as
`isposdef(x::Number) = imag(x)==0 && real(x) > 0`
so it makes sense for us to stick to that "strict" definition here too.